### PR TITLE
print_summary() and get_txt_annotation() enhancement; adding print_predefined_models()

### PR DESCRIPTION
**Pull request by wenyushi**
The push includes several features:
- enhance print_summary() and it will provide more info like memory cost and FLOPS. 
- enhance get_txt_annotation() in utils.py. After the push, users can generate txt format aligned with open source YOLO package. 
- adding print_predefined_models in utils.py which will print all of the predefined models in dlpy.application module.

### CODE REVIEW

File: dlpy/utils.py [hunk 1]
```diff
@@ -1123,6 +1123,11 @@ def _convert_xml_annotation(filename, coord_type, resize):
     if object_ is None:
         in_file.close()
         return
+    # write in all classes
+    if name_file:
+        with open(name_file, 'r') as nf:
+            line = nf.readlines()
+            cls_list = [l.strip() for l in line]
```
Comment (ethem-kinginthenorth) [inline dlpy/utils.py]: if name_file is not there, this part would not complain as it is in the with,

File: dlpy/utils.py [hunk 2]
```diff
@@ -1243,6 +1257,17 @@ def get_txt_annotation(local_path, coord_type, image_size = (416, 416), label_fi
         Specifies the list of filename with XML extension under local_path to be parsed.
         If label_files is not specified, all of XML files under local_path will be parsed .
         Default: None
+    name_file : str, optional
+        Specifies the path of name_file which lists all of the names for the classes in your dataset.
+        If you specify the option, the function will generate txt using index to represent category
+        and space as separator. For example:
+            17 0.539766 0.492976 0.317406 0.345796
+            17 0.557742 0.265619 0.083922 0.087191
+        Otherwise, it will generate the format can be consumed by SAS platform.
```
Comment (ethem-kinginthenorth) [inline dlpy/utils.py]: can we have the numbers the same, and indicate that 17 is the internal number assigned for the person category

File: dlpy/datasources/coco.names [hunk 3]
```diff
@@ -0,0 +1,4 @@
+White Rook
+White King
+Black King
+White Kinghuh
```
[the remark this answers is not in the document]
Comment (wenyushi) [reply dlpy/datasources/coco.names]: a fake class name

File: dlpy/layers.py [hunk 4]
```diff
@@ -326,12 +327,28 @@ def summary(self):
             stride = self.stride
         else:
             stride = self.config.get('stride', '')
+        # get feature memory of a layer
+        self.output_mem = multiply_elements(self.output_size) \
```
[the remark this answers is not in the document]
Comment (wenyushi) [reply dlpy/layers.py]: output_mem represents how many elements in a layer's feature map. Before, we only have a column to say the feature map size of a layer. I want to use the variable to calculate how many memory the model occupies.

File: dlpy/network.py [hunk 5]
```diff
@@ -649,14 +651,33 @@ def print_summary(self):
...
-                                              'Output Size', 'Number of Parameters'])
-                display(pd.concat([self.summary, total], ignore_index = True))
+                total_FLOPS = 0
+                total_output_mem = 0
+                for l in self.layers:
+                    total_FLOPS += l.FLOPS if l.FLOPS else 0
+                    total_output_mem += l.output_mem if l.output_mem else 0
+
+                MB = 1 << 20
+                # assuming single float
```
[the remark this answers is not in the document]
Comment (wenyushi) [reply dlpy/network.py]: yes

File: dlpy/network.py [hunk 6]
```diff
@@ -649,14 +651,33 @@ def print_summary(self):
...
-                                     columns=['Layer Id', 'Layer', 'Type', 'Kernel Size', 'Stride', 'Activation',
-                                              'Output Size', 'Number of Parameters'])
-                display(pd.concat([self.summary, total], ignore_index = True))
+                total_FLOPS = 0
+                total_output_mem = 0
+                for l in self.layers:
+                    total_FLOPS += l.FLOPS if l.FLOPS else 0
+                    total_output_mem += l.output_mem if l.output_mem else 0
+
+                MB = 1 << 20
```
[the remark this answers is not in the document]
Comment (wenyushi) [reply dlpy/network.py]: I switch to 2^20

File: dlpy/tests/test_applications.py [hunk 7]
```diff
@@ -629,6 +629,9 @@ def test_mobilenetv1(self):
         self.assertTrue(len(model.layers) == 57)
         self.assertTrue(model.layers[49]._output_size == (7, 7, 2048))
         model.print_summary()
+        self.assertEqual(model.total_FLOPS_in_unit, 20932.283)
+        self.assertEqual(model.total_parameters_in_unit, 483.339)
+        self.assertEqual(model.total_output_mem_in_unit, 342.816)
```
[the remark this answers is not in the document]
Comment (wenyushi) [reply dlpy/tests/test_applications.py]: It doesn't match with tkdl. The snippet only calculates model weights and feature maps cost.

File: dlpy/utils.py [hunk 8]
```diff
@@ -1110,7 +1110,7 @@ def _convert_coco(size, box, resize):
     return (x_min, y_min, x_max, y_max)
 
 
-def _convert_xml_annotation(filename, coord_type, resize):
+def _convert_xml_annotation(filename, coord_type, resize, name_file = None):
```
[the remark this answers is not in the document]
Comment (wenyushi) [reply dlpy/utils.py]: I put the comments at get_txt_annotations() as _convert_xml_annotation() is hidden.

**Pull request merged**